app/access/service_key_access.rb

```diff
@@ -14,5 +14,11 @@ def read?(service_key)
       return true if admin_user?
       service_key.service_instance.space.developers.include?(context.user)
     end
+
+    def index?(_, params=nil)
+      return true if admin_user?
+      return true unless params && params.key?(:related_obj)
```

Comment:
@rmasand : the basic model controller is using validate_access for associated model of service instance to check if the current user has access to the endpoint (/v2/service_instances/:guid/service_keys) in enumerate_related method at [117](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/controllers/base/model_controller.rb#L117). The service instance object will be passed to `params` of `index?` method through line [129](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/controllers/base/model_controller.rb#L129). But the index method will also apply the existing service key endpoint  (/v2/service_keys). So for this case, it will be validate with true value and leave it to be checked by user_visibility at line [134](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/controllers/base/model_controller.rb#L134). This is expected API behavior to return 0 results. Does it make sense? Thanks for your comments.